### ISSUE TITLE
postBuildStep's should probably be scheduled regardless of build skips

### ISSUE THREAD

**Issue by antifuchs** [nix-community/buildbot-nix#295]
I have an "effect" (not a hercules-ci-effect! no idea how to use those tbh) via a `postBuildStep` that I would like to run only on the default branch of my repo. If I make a pull request that affects that attribute, it properly gets run on the PR branch and decides it has nothing to do; but if I merge the PR, the build is skipped and the postBuildStep isn't executed - it isn't even scheduled.

This is what a build on a PR branch looks like:
<img width="849" alt="image" src="https://github.com/user-attachments/assets/d5f0b0fd-606a-4a26-bce5-965652734c9d">

And this is on the merge commit:
<img width="850" alt="image" src="https://github.com/user-attachments/assets/35d06d34-707c-475b-9d0d-aab4e9e84d1d">

I believe what happens here is that the evaluation counts as "cached" coming out of nix-eval-jobs, which prompts buildbot-nix to not build & then not schedule the following postBuildSteps either.

What I think would be neat is if we could:
* schedule `postBuildSteps`, either by default or via a configurable attribute
* maybe expose the `doStepIf` buildbot attribute, so steps can decide for themselves based on the properties of the build whether they are supposed to run or not.

**Comment by Mic92** [comment]
I think these postBuildSteps should be just used for custom binary cache logic rather than continuous delivery and I rather not want to make building individual packages more expensive by adding more steps since it's already quite slow. This would be than only available when we merged: https://github.com/nix-community/buildbot-nix/pull/68

**Comment by Mic92** [comment]
If you  want to make Hercules Ci happen faster, you can probably help us implementing it.

**Comment by antifuchs** [comment]
Yeah, I can see how abusing the "build" metaphor isn't something you would want to do. That said, I can't wrap my head around how hercules-ci effects works, and how it's supposed to fit into buildbot-nix. Is there a design doc I can look at?

**Comment by antifuchs** [comment]
I get that - but how will hci effects integrate into buildbot? Will a worker act as the equivalent of the hercules-ci agent, or will that still require the agent to be in place? Probably should take that discussion to #68 (:

**Comment by Mic92** [comment]
I already implemented the runner interface of hercules-ci effects, so it will run in a buildbot worker as in a bubblewrap container.

**Comment by Mic92** [comment]
https://github.com/nix-community/buildbot-nix/blob/3ac54ba1cc53aae01dd27055a710fbebab19acfe/buildbot_effects/__init__.py#L157

**Comment by Mic92** [comment]
https://github.com/nix-community/buildbot-nix/pull/68/files#diff-df8c266d76f942a320d71b583a24da5fa8ecd8135993a696f376dbd960359be7R1264
